designs/credentialManagement/credentialManagement.md

```diff
@@ -0,0 +1,139 @@
+# Credential Management
+
+## Abstract
+
+This document describes the process of managing and switching of AWS credentials within the toolkit.
+
+## Motivation
+
+AWS provides many different ways to retrieve the credentials required to make calls to AWS including but not limited to:
+credential/config files, environment variables, system properties, HTTP metadata APIs. A user may also assume roles using 
+a source set of credentials or use MFA. Due to this large matrix of possibilities the toolkit strives to provide an 
+intuitive abstraction layer on top of some of these sources so that the implementation details are hidden and a 
+consistent user experience can be provided, while also enabling future expansion of new sources, including possibly by 
+third party plugins.
+
+## Classes and Concepts
+![ClassDiagram]
+
+1. `AwsRegion` - Value class that represents an AWS Region and joins together related data for that region. This data is sourced from the endpoints.json file.
```

Comment:
nit: Oh no Scala terminology, I think `Data class` is more wildly understood as well

Reply:
Sure